### ISSUE TITLE
Overloading multiple signatures for a single ref

### ISSUE THREAD

**Issue by soulitzer** [pytorch/pytorch#81959]
Consider an op like `arange` which is able to support the following two signatures using python arg parser

https://github.com/pytorch/pytorch/blob/8d0cbce0696e76c9e4c929c24c0331df25125b40/torch/csrc/autograd/python_torch_functions_manual.cpp#L87-L90

It seems tricky to add an op like this since Python does not support overloads, and arguments with defaults must come after arguments that don't have defaults.  

but one way to avoid only being able to support one of the overloads could be to accept a more general signature and parse it myself:
  - in the below example, we simply allow `arange(arg1: optional[Number], arg2: optional[Number], step=1, *, ...)`  add some checks to route to the correct prim, which does not take any optional arguments
  - registering stubs with `@overload` decorator from the `typing` module seems to help with static checking tools 

https://github.com/pytorch/pytorch/blob/d1a9372e611b6e833f83fab57816819b7732e1cc/torch/_refs/__init__.py#L2989-L3055

Curious if others think this is a good general solution

**Comment by ezyang** [comment]
Yes, this is one of the more painful things for PrimTorch refs, as you don't get the argument parsing that PyTorch has built in. What you suggest is as about as good as it gets, barring actually building an overload resolver in Python.
